kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorNetworkIT.java

```diff
@@ -0,0 +1,335 @@
+    setupConnectorProps();
+  }
+
+  @After
+  public void after(){
+    emptyBucket(TEST_BUCKET_NAME);
+  }
+
+  @AfterClass
+  public static void deleteBucket() {
+    // delete the test bucket
+    deleteBucket(TEST_BUCKET_NAME);
+  }
+
+  /**
+   * Success scenario : End to end test
+   * @throws InterruptedException
+   */
+  @Test
+  @Ignore
+  public void testToAssertConnectorAndDestinationRecords() throws InterruptedException {
+
+    // create topics in Kafka
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+    // send records to kafka
+    sendRecordsToKafka(NUM_RECORDS_PRODUCED);
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    int expectedFileCount = NUM_RECORDS_PRODUCED / FLUSH_SIZE;
+    waitForFilesInBucket(TEST_BUCKET_NAME, expectedFileCount);
+
+    // assert files in bucket
+    List<String> expectedFilenames = getExpectedFilenames(TEST_TOPIC_NAME, EXPECTED_PARTITION,
+        FLUSH_SIZE, NUM_RECORDS_PRODUCED, EXPECTED_FILE_EXTENSION);
+    assertTrue(fileNamesValid(TEST_BUCKET_NAME, expectedFilenames));
+  }
+
+  /**
+   * Case in which bucket permissions/policies are changed while uploading records
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   * @throws InterruptedException
+   */
+  @Test
+  @Ignore
+  public void testWithRevokedWritePermissions() throws InterruptedException {
+
+    addReadWritePolicyToBucket(TEST_BUCKET_NAME);
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+    // send records to kafka
+    sendRecordsToKafka(NUM_RECORDS_PRODUCED);
+
+    props.put("aws.access.key.id", System.getenv("SECONDARY_USER_ACCESS_KEY_ID"));
+    props.put("aws.secret.access.key", System.getenv("SECONDARY_USER_SECRET_ACCESS_KEY"));
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    int expectedFileCount = NUM_RECORDS_PRODUCED / FLUSH_SIZE;
+    waitForFilesInBucket(TEST_BUCKET_NAME, expectedFileCount);
+    // assert files in bucket
+    List<String> expectedFilenames = getExpectedFilenames(TEST_TOPIC_NAME, EXPECTED_PARTITION,
+        FLUSH_SIZE, NUM_RECORDS_PRODUCED, EXPECTED_FILE_EXTENSION);
+    assertTrue(fileNamesValid(TEST_BUCKET_NAME, expectedFilenames));
+
+    // revoke read/write permission
+    alterReadWritePolicyOfBucket(TEST_BUCKET_NAME);
+    /*
+     Intentional sleep added in order for the bucket permission to be altered to read only to
+     come into affect.
+    */
+    Thread.sleep(10000);
+    // produce more records to kafka
+    sendRecordsToKafka(NUM_RECORDS_PRODUCED);
+
```

Comment:
Should we have a short sleep here after sending records to make sure files don't arrive?

Reply:
Added a 10s sleep.